### ISSUE TITLE
Generating final.ext.raw from final.raw for speaker models

### ISSUE THREAD

**Issue by mac-and-cheese** [alphacep/vosk-api#171]
Hi,
the provided speaker model https://alphacephei.com/vosk/models/vosk-model-spk-0.3.zip contains two files (final.ext.raw and mfcc.conf). 

I am trying to figure out how get the file final.ext.raw from the kaldi model http://kaldi-asr.org/models/m6. 

I tried using final.raw present in the kaldi m6 model as final.ext.raw, but this gives a 5164 dimension output instead of the expected 128 dim vector.

Any info regarding this is appreciated, thank you.

**Comment by nshmyrev** [comment]
You need to extract lower layers which is actually used for xvectors, like here:

`nnet3-copy --nnet-config=extract.config final.raw final.final.raw`

as in here: https://github.com/kaldi-asr/kaldi/blob/f88d5a361d0d19bf33fd344df79f634fa9eb71ff/egs/sre08/v1/sid/nnet3/xvector/extract_xvectors.sh#L58

**Comment by mac-and-cheese** [comment]
@nshmyrev Thank you!

**Comment by ChinmayPatil11** [comment]
Hello @nshmyrev , @mac-and-cheese ,
I'm new to speaker model.
I also had the same issue but haven't been able to figure it out. Can you point out to a resource or note the steps how to extract the final.raw model to final.ext.raw. Any help would be appreciated. Thank you!

**Comment by nshmyrev** [comment]
@ChinmayPatil11 the command is in the comment above

**Comment by ChinmayPatil11** [comment]
Hi @nshmyrev , I tried using this command but I have the following error

> ERROR (nnet3-xvector-compute[5.5]:Input():kaldi-io.cc:756) Error opening input stream 'nnet3-copy --nnet-config=extract.config final.raw final.final.raw'

The current neural network configuration is as follows

> please note that it is important to have input layer with the name=input
> The frame-level layers
> input dim=${feat_dim} name=input
> relu-batchnorm-layer name=tdnn1 input=Append(-2,-1,0,1,2) dim=512
> relu-batchnorm-layer name=tdnn2 input=Append(-2,0,2) dim=512
> relu-batchnorm-layer name=tdnn3 input=Append(-3,0,3) dim=512
> relu-batchnorm-layer name=tdnn4 dim=512
> relu-batchnorm-layer name=tdnn5 dim=1500
> The stats pooling layer. Layers after this are segment-level.
> In the config below, the first and last argument (0, and ${max_chunk_size})
> means that we pool over an input segment starting at frame 0
> and ending at frame ${max_chunk_size} or earlier.  The other arguments (1:1)
> mean that no subsampling is performed.
> stats-layer name=stats config=mean+stddev(0:1:1:${max_chunk_size})
> This is where we usually extract the embedding (aka xvector) from.
> relu-batchnorm-layer name=tdnn6 dim=128 input=stats
> This is where another layer the embedding could be extracted
> from, but usually the previous one works better.
> CHANGES/FIXME:output dimensions changed from 512 to 128 in last two layers
> relu-batchnorm-layer name=tdnn7 dim=128
> output-layer name=output include-log-softmax=true dim=${num_targets}

Thank you.

**Comment by nshmyrev** [comment]
> Hi @nshmyrev , I tried using this command but I have the following error

You have to run it separately, not within the extractor script.

**Comment by vonguyen1982** [comment]
@nshmyrev I am on windows OS so it is hard to run the command. Do you mind to drop the final.ext.raw some where to download. 
IN model I don't have final.ext.raw.
ERROR (VoskAPI:Input():kaldi-io.cc:756) Error opening input stream 'C:\Demo\VoskLiveServer\model\final.ext.raw'

I am using the https://alphacephei.com/vosk/models/vosk-model-en-us-0.22.zip

**Comment by nshmyrev** [comment]
@vonguyen1982 it is a speaker model, inside https://alphacephei.com/vosk/models/vosk-model-spk-0.4.zip
